kratos/python/add_data_communicator_to_python.cpp

```diff
@@ -0,0 +1,60 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main author:     Jordi Cotela
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "add_data_communicator_to_python.h"
+#include "includes/define_python.h"
+#include "includes/data_communicator.h"
+
+namespace Kratos {
+
+namespace Python {
+
+void AddDataCommunicatorToPython(pybind11::module &m)
+{
+    namespace py = pybind11;
+
+    py::class_<DataCommunicator, DataCommunicator::Pointer>(m,"DataCommunicator")
+    .def("Barrier", &DataCommunicator::Barrier)
+    .def("Sum", (int (DataCommunicator::*)(const int, const int) const) &DataCommunicator::Sum)
```

Comment:
for those functions we should decide if we append the datatype
=> anyway needed / recommended for e.g. scatter as we saw :)

Reply:
@philbucher I have added vector versions in the python interface for these methods in the latest commit. Please take a look, since I am rather unconvinced by the naming convention I used...

Also, I am using "native" pybind11 list <-> std::vector conversions and I have the feeling that it takes ages to compile/link. Did you also notice this for the current mpi python?

Reply:
what does this syntax mean?